src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/SocketsHttpHandlerConfigurationTest.cs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET5_0_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Security;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Extensions.Http
+{
+    public class SocketsHttpHandlerConfigurationTest
+    {
+        private const string ParentSectionName = "HttpClientSettings";
+        private const string SectionName = "ConfiguredByIConfiguration";
+        private static Dictionary<string, string> s_configContent = new Dictionary<string, string>
+        {
+            { $"{ParentSectionName}:{SectionName}:AllowAutoRedirect", "true" },
+            { $"{ParentSectionName}:{SectionName}:UseCookies", "false" },
+            { $"{ParentSectionName}:{SectionName}:ConnectTimeout", "00:00:05" },
+            { $"{ParentSectionName}:{SectionName}:PooledConnectionLifetime", "00:01:00" },
```

Comment:
Would it make sense to cover all the settings?

Reply:
https://github.com/dotnet/runtime/pull/89202